packages/synapse-react-client/src/components/Authentication/LoginForm.tsx

```diff
@@ -33,11 +33,22 @@ type Props = {
   twoFactorAuthResetUri?: string
 }
 
+function appendTwoFAResetTokenQueryParam(url: URL): string {
+  const queryParam = 'twoFAResetToken'
+
+  if (url.searchParams.size === 0) {
+    // No params, so use `?`
+    return `${url.toString()}?${queryParam}=`
+  } else {
+    // Already has params, so use '&'
+    return `${url.toString()}&${queryParam}=`
+  }
+}
+
 export default function LoginForm(props: Props) {
   const defaultRegistrationUrl = useOneSageURL('/register1')
-  const defaultTwoFactorAuthResetUrl = useOneSageURL(
-    '/reset2FA?twoFAResetToken=',
-  )
```

Comment:
Previously this appended the appId, which would end up causing the twoFAResetToken to be in the wrong place after the server appended the token.